src/util/system.cpp

```diff
@@ -294,17 +293,18 @@ static bool InterpretNegatedOption(std::string& key, std::string& val)
         ++option_index;
     }
     if (key.substr(option_index, 2) == "no") {
-        bool bool_val = InterpretBool(val);
+        const bool bool_val = InterpretBool(val);
         key.erase(option_index, 2);
         if (!bool_val ) {
             // Double negatives like -nofoo=0 are supported (but discouraged)
             LogPrintf("Warning: parsed potentially confusing double-negative %s=%s\n", key, val);
```

Comment:
(https://github.com/bitcoin/bitcoin/pull/16097/commits/e0d187dfeb18b026de22bd7960b2a50c2b958e1a)
 
FWIW, this results in a slightly incorrect error message:
```
$ ./src/bitcoind -datadir=/data/bitcoin -noconnect=0 -logthreadnames

2019-08-01T16:03:23Z Warning: parsed potentially confusing double-negative -connect=0
```
which I think should read "double-negative -noconnect=0".

Reply:
> I think should read "double-negative -noconnect=0".

You are right. This remained untouched to preserve the old behavior and make this PR refactoring only. Details: https://github.com/bitcoin/bitcoin/pull/16097#discussion_r308829507 and https://github.com/bitcoin/bitcoin/pull/16097#discussion_r309281546.